### PR TITLE
PTTP 2971/use EKS module for cluster creation

### PR DESCRIPTION
**Pull request by MTouhid**
Using the EKS module to generate the entire EKS control plane.
Previously we were only creating EKS cluster but that was a lot more involved. Use of this module made it simpler.

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=432&projectKey=PTTP&modal=detail&selectedIssue=PTTP-2971&quickFilter=1066

### CODE REVIEW

**Review by wanieldilson — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,30 @@
...
+	aws-vault clear && aws-vault exec $$AWS_VAULT_PROFILE -- terraform plan
+
+apply:
+	aws-vault clear && aws-vault exec $$AWS_VAULT_PROFILE --duration=2h -- terraform apply
+
+destroy:
+	aws-vault clear && aws-vault exec $$AWS_VAULT_PROFILE -- terraform destroy
+
+clean:
+	rm -rf .terraform/ terraform.tfstate*
```
Comment (thip) [inline Makefile:25]: 🎉

File: modules/monitoring_platform/eks.tf [hunk 2]
```diff
@@ -1,48 +1,24 @@
...
+  vpc_id  = aws_vpc.main.id
 
-resource "aws_iam_role_policy_attachment" "service_role_policy_attachment" {
-  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSServicePolicy"
-  role       = aws_iam_role.cluster_role[0].name
-  count      = var.is_eks_enabled ? 1 : 0
+  worker_groups = [
+    {
+      name                 = "prometheus-worker-group"
+      instance_type        = "t3.small"
```
Comment (thip) [inline modules/monitoring_platform/eks.tf:20]: Probably worth going bigger - take a look at the amount of cpu/memory being used by thanos components in fargate. Thanos store is pretty beefy so the worker should be at least as powerful as what that needs.

File: modules/monitoring_platform/eks.tf [hunk 3]
```diff
@@ -1,48 +1,24 @@
...
+      asg_desired_capacity = 2
```
Comment (thip) [inline modules/monitoring_platform/eks.tf:21]: Assuming this is the number of nodes in the cluster, I *think* you need at least 3 for a HA setup

**Review by thip — APPROVED** [review]
- Left some comments about cluster/node sizing although you don't have to worry about it now, will need to tune it later though.
- should modules/prometheus/eks.tf be deleted?

Otherwise lgtm, I like the makefile and the way you're picking up the aws-vault profile from .env 🎉

**Pull request merged**